client/checkout/classic/index.js

```diff
@@ -45,12 +44,29 @@ jQuery( function ( $ ) {
 			} );
 		}
 	);
+
+	const fontRules = [],
+		sheets = document.styleSheets,
+		fontDomains = [
+			'fonts.googleapis.com',
+			'fonts.gstatic.com',
+			'fast.fonts.com',
+			'use.typekit.net',
+		];
+	for ( let i = 0; i < sheets.length; i++ ) {
+		if ( ! sheets[ i ].href ) {
+			continue;
+		}
+		const url = new URL( sheets[ i ].href );
+		if ( -1 !== fontDomains.indexOf( url.hostname ) ) {
+			fontRules.push( {
+				cssSrc: sheets[ i ].href,
+			} );
+		}
+	}
+
 	const elements = api.getStripe().elements( {
-		fonts: [
-			{
-				cssSrc: upeFontSource,
-			},
-		],
+		fonts: fontRules,
```

Comment:
(minor) it looks like the approach could be shortened slightly with `document.querySelectorAll`:
```
const fontDomains = [
	'fonts.googleapis.com',
	'fonts.gstatic.com',
	'fast.fonts.com',
	'use.typekit.net',
];

const elements = api.getStripe().elements( {
	fonts: Array.from(
		document.querySelectorAll(
			fontDomains
				.map(
					( domain ) => `link[rel="stylesheet"][href*="${ domain }"]`
				)
				.join( ', ' )
		)
	).map( ( node ) => node.href ),
} );
```

Both approaches seem to be working 👍

Reply:
A quick test with console.time() shows my approach is 0.1 ms faster so I'll stick to it

Reply:
Good thinking on the timing!
In my tests I had an average of 0.048ms difference over 15 tests on Storefront.  But on other themes, the difference was even minor. I'm guessing it depends on how many styles domains there are on the page - the more styles there are, the closer the gap.